main.go

```diff
@@ -185,17 +225,23 @@ func (handler *BuildHandler) ServeHTTP(
 
 	runtime.LockOSThread()
 
-	err = rawSetuid(handler.buildUid)
+	err = rawSetuid(handler.buildUID)
 	if err != nil {
 		response.WriteHeader(http.StatusInternalServerError)
 		fmt.Fprintf(
-			topLevelLogger, "can't set uid to %d: %s", handler.buildUid, err,
+			topLevelLogger, "can't set uid to %d: %s", handler.buildUID, err,
 		)
 		return
 	}
 
+	buildInfo := BuildInfo{
+		repository: repoURL,
+		startedAt:  time.Now(),
+		status:     "in progress",
+	}
+	buildInfoPointer := handler.saveNewBuild(buildInfo)
```

Comment:
Now, just pass pointer to the `buildInfo` around (`buildInfo := &buildInfo{...}`). Therefore, it will be no need for returning `buildInfoPointer` and no need for assignment at the 261 line.